src/Components/LoginForm.tsx

```diff
@@ -0,0 +1,233 @@
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [newUser, setNewUser] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [emailDisabled, setEmailDisabled] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showPrevUsers, setShowPrevUsers] = useState(false);
+  const hasErrorFromServer = !!errorMessage;
+
+  const ref = useRef<HTMLInputElement>(null);
+
+  let tempEmail = '';
+
+  const prevUsers: string[] | 0 = JSON
+    .parse(localStorage.getItem('prevusers') || '0');
+
+  const toshowPrevUsers = useMemo(() => {
+    return prevUsers !== 0 ? prevUsers
+      .filter(preMail => {
+        return email !== '' ? preMail.includes(email) : true;
+      }) : 0;
+  }, [email]);
+
+  const clearNotification = () => {
+    setErrorMessage('');
+  };
+
+  const newUserObject = {
+    name,
+    username: '',
+    email,
+    phone: '',
+    website: '',
+  };
+
+  const addUser = async (e: FormEvent) => {
+    e.preventDefault();
+    setIsLoading(true);
+
+    const exitingUserLogin = async (emailForLogin: string) => {
+      try {
+        const userExits = await findUser(emailForLogin);
+
+        if (userExits.length !== 0) {
+          localStorage.setItem(
+            'user',
+            JSON.stringify(userExits[0]),
+          );
+          setUser(userExits[0]);
+          addPrevUser(emailForLogin);
+        } else {
+          setNewUser(true);
+          setEmailDisabled(true);
+          setIsLoading(false);
+        }
+      } catch {
+        setErrorMessage('Unable to login');
+        setIsLoading(false);
+      }
+    };
+
+    if (email && !newUser) {
+      exitingUserLogin(email);
+    }
+
+    if (name && newUser) {
+      try {
+        const addedUser = await addUsers(newUserObject);
+
+        if (addedUser) {
+          exitingUserLogin(email);
+        }
+      } catch {
+        setErrorMessage('Unable to login');
+        setIsLoading(false);
+      }
+    }
+  };
+
+  const selectPrevUsers = (prevmail: string) => {
+    if (ref.current) {
+      ref.current.focus();
+    }
+
+    setEmail(prevmail);
+    setShowPrevUsers(false);
+    tempEmail = prevmail;
+  };
+
+  return (
+    <>
+      <form
+        className="box is-size-3 mt-5"
+        onSubmit={(e) => addUser(e)}
+      >
+        <h1 className="title is-3">Log in to open todos</h1>
+        <div className="field prevuser-p">
+          <label className="label" htmlFor="user-email">Email</label>
+          <div className="control has-icons-left">
+            <input
+              type="email"
+              id="user-email"
+              className="input"
+              placeholder="Enter your email"
+              value={email}
+              onChange={(e) => {
+                setEmail(e.target.value);
+                if (tempEmail !== e.target.value) {
+                  setShowPrevUsers(true);
+                }
+              }}
```

Comment:
Don't write functions inside JSX. It's not a good practice. Create a separate method